fastcrypto-tbls/src/nodes.rs

```diff
@@ -135,41 +135,47 @@ impl<G: GroupElement + Serialize> Nodes<G> {
         hash.finalize()
     }
 
-    /// Reduce weights up to an allowed delta in the original total weight.
+    /// Create a new set of nodes. Nodes must have consecutive ids starting from 0.
+    /// Reduces weights up to an allowed delta in the original total weight.
     /// Finds the largest d such that:
     /// - The new threshold is ceil(t / d)
     /// - The new weights are all divided by d (floor division)
     /// - The precision loss, counted as the sum of the remainders of the division by d, is at most
     ///   the allowed delta
     /// In practice, allowed delta will be the extra liveness we would assume above 2f+1.
     ///
-    /// total_weight_lower_bound allows limiting the level of reduction (e.g., in benchmarks). To get the best results,
-    /// set it to 1.
-    ///
-    /// Assumption: reduce is called only once per a set of parameters (and not repeatedly).
-    pub fn reduce(&self, t: u16, allowed_delta: u16, total_weight_lower_bound: u16) -> (Self, u16) {
-        assert!(total_weight_lower_bound <= self.total_weight && total_weight_lower_bound > 0);
+    /// total_weight_lower_bound allows limiting the level of reduction (e.g., in benchmarks). To
+    /// get the best results, set it to 1.
+    pub fn new_reduced(
+        nodes_vec: Vec<Node<G>>,
+        t: u16,
+        allowed_delta: u16,
+        total_weight_lower_bound: u16,
+    ) -> FastCryptoResult<(Self, u16)> {
+        let n = Self::new(nodes_vec)?; // checks the input, etc
+
+        assert!(total_weight_lower_bound <= n.total_weight && total_weight_lower_bound > 0);
```

Comment:
Is it safe to do this as an assertion?

Reply:
yes, those parameters should never not satisfy those conditions